app/controllers/incomeSources/add/CheckUKPropertyDetailsController.scala

```diff
@@ -97,23 +100,42 @@ class CheckUKPropertyDetailsController @Inject()(val checkUKPropertyDetails: Che
         }
   }
 
-  def handleRequest(isAgent: Boolean)(implicit user: MtdItUser[_]): Future[Result] = {
+  def handleRequest(isAgent: Boolean)(implicit user: MtdItUser[_]): Future[Result] = withIncomeSourcesFS {
     val backUrl = getBackUrl(isAgent)
     val postAction = getSubmitUrl(isAgent)
     val errorHandler = getErrorHandler(isAgent)
 
-    withIncomeSourcesFS {
-      getUKPropertyDetailsFromSession(sessionService)(user, ec).map(_.toOption).map {
-        case Some(checkUKPropertyViewModel: CheckUKPropertyViewModel) =>
-          Ok(
-            checkUKPropertyDetails(viewModel = checkUKPropertyViewModel,
-              isAgent = isAgent,
-              backUrl = backUrl,
-              postAction = postAction))
-        case None => Logger("application").error(
-          s"[CheckUKPropertyDetailsController][handleRequest] - Error: Unable to build UK property details")
-          errorHandler.showInternalServerError()
-      }
+    getUKPropertyDetailsFromSession(user, ec).map(_.toOption).map {
+      case Some(checkUKPropertyViewModel: CheckUKPropertyViewModel) =>
+        Ok(
+          checkUKPropertyDetails(viewModel = checkUKPropertyViewModel,
+            isAgent = isAgent,
+            backUrl = backUrl,
+            postAction = postAction))
+      case None => Logger("application").error(
+        s"[CheckUKPropertyDetailsController][handleRequest] - Error: Unable to build UK property details")
+        errorHandler.showInternalServerError()
+    }
+  }
+
+  def getUKPropertyDetailsFromSession(implicit user: MtdItUser[_], ec: ExecutionContext): Future[Either[Throwable, CheckUKPropertyViewModel]] = {
+    for {
+      startDate <- sessionService.getMongoKeyTyped[LocalDate](dateStartedField, JourneyType(Add, UkProperty))
+      accMethod <- sessionService.getMongoKeyTyped[String](incomeSourcesAccountingMethodField, JourneyType(Add, UkProperty))
```

Comment:
This could be improved by using the getMongo method, requires only 1 call to get all journey data (this doesn't need to be done now)